src/websocket.rs

```diff
@@ -42,6 +56,91 @@ impl AdminWebsocket {
+        match env::var("DEVICE_BUNDLE_PASSWORD") {
+            Ok(pass) => Some(pass),
+            _ => None,
+        }
+    }
+
+    fn mem_proof_server_url(&mut self) -> String {
+        match env::var("MEM_PROOF_SERVER_URL") {
+            Ok(url) => url,
+            _ => "https://test-membrane-proof-service.holo.host".to_string(),
+        }
+    }
+
+    async fn try_registration_auth(
+        &mut self,
+        config: &Config,
+        holochain_public_key: PublicKey,
+    ) -> Fallible<()> {
+        match config {
+            Config::V2 {
+                registration_code,
+                settings,
+                ..
+            } => {
+                let email = settings.admin.email.clone();
+                let payload = Registration {
+                    registration_code: registration_code.clone(),
+                    agent_pub_key: holochain_public_key,
+                    email: email.clone(),
+                    payload: RegistrationPayload {
+                        role: "host".to_string(),
+                    },
+                };
+                let mem_proof_server_url =
+                    format!("{}/register-user/", self.mem_proof_server_url());
+                let resp = CLIENT
+                    .post(mem_proof_server_url)
+                    .json(&payload)
+                    .send()
+                    .await?;
+                match resp.error_for_status_ref() {
+                    Ok(_) => {
+                        let reg: RegistrationRequest = resp.json().await?;
+                        println!("Registration completed message ID: {:?}", reg);
+                        // save mem-proofs into a file on the hpos
+                        let mut file = File::create(mem_proof_path())?;
+                        file.write_all(reg.mem_proof.as_bytes())?;
+                    }
+                    Err(_) => {
+                        let err: RegistrationError = resp.json().await?;
+                        send_failure_email(email, err.to_string()).await?;
+                        return Err(AuthError::RegistrationError(err.to_string()).into());
+                    }
+                }
+            }
+            Config::V1 { settings, .. } => {
+                send_failure_email(
+                    settings.admin.email.clone(),
+                    AuthError::ConfigVersionError.to_string(),
+                )
+                .await?;
+                return Err(AuthError::ConfigVersionError.into());
+            }
+        }
+        Ok(())
+    }
+
+    async fn enable_memproof_dev_net(&mut self) -> Fallible<()> {
+        let config = self.get_hpos_config()?;
+        let password = self.device_bundle_password();
+        let holochain_public_key =
+            hpos_config_seed_bundle_explorer::holoport_public_key(&config, password).await?;
+        // Get mem-proof by registering on the ops-console
+        self.try_registration_auth(&config, holochain_public_key)
```

Comment:
I don't understand why we're doing any of this. Configure-holochain shouldn't need to talk to the memproof server itself?

Reply:
@alastairong 
From what Joel told me. Membrane proof can't get the generated key on dev net because it is created after it requests membrane proof. So Joel said the simplest solution is probably to request the membrane proof (only for devNet) after the key is geenrated which is in `configure holochain`

Reply:
Oh jeez now I get it. Wow this is really brute forcing it huh?

So in devNet require requesting a second memproof with the real agentId in configure holochain?

Reply:
It is not brute-forcing it because its should only generate it when the random key was created